asset-transfer-secured-agreement/application-gateway-typescript/src/contractWrapper.ts

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright IBM Corp. All Rights Reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import { Contract } from '@hyperledger/fabric-gateway';
+import { TextDecoder } from 'util';
+import { Asset, AssetJSON, AssetPrice, AssetPriceJSON, AssetPrivateData, AssetProperties, AssetPropertiesJSON, GREEN, parse, RED, RESET } from './utils';
+
+export class ContractWrapper {
+
+    readonly #contract: Contract;
+    readonly #org: string;
+    readonly #utf8Decoder  = new TextDecoder();
+    readonly #randomBytes: string;
+
+    public constructor(contract: Contract, org: string, randomBytes: string) {
+        this.#contract = contract;
+        this.#org = org;
+        this.#randomBytes = randomBytes;
+    }
+
+    public async createAsset(asset: Asset, privateData: AssetPrivateData): Promise<void> {
+        console.log(`${GREEN}--> Submit Transaction: CreateAsset, ${asset.AssetId} as ${asset.OwnerOrg} - endorsed by Org1${RESET}`);
+        try {
+            const assetPropertiesJSON: AssetPropertiesJSON = Object.assign({}, {objectType: 'asset_properties',
+                assetID: asset.AssetId,
+                color: privateData.Color,
+                size: privateData.Size,
+                salt: this.#randomBytes });
+
+            await this.#contract.submit('CreateAsset', {
+                arguments: [asset.AssetId, asset.PublicDescription],
+                transientData: { asset_properties: JSON.stringify(assetPropertiesJSON)},
+            });
+
+            console.log(`*** Result: committed, asset ${asset.AssetId} is owned by Org1`);
+        }catch (e) {
+            console.log(`${RED}*** Failed: createAsset - ${e}${RESET}`);
+        }
```

Comment:
I don't really like this pattern of catching, logging and then ignoring for all transaction errors. It makes it hard to see if the application is actually working correctly because it never actually fails. I would be tempted to let these methods throw if the transaction invocation fails, then try/catch around the call in the app main() function for only the invocation that are expected to fail because they are demonstrating a failure condition.